qemu/tests/qemu_guest_agent.py

```diff
@@ -159,26 +159,50 @@ def gagent_uninstall(self, session, vm, *args):
     @error_context.context_aware
     def gagent_start(self, session, vm, *args):
         """
-        Start qemu-guest-agent in guest,if start a running service,for rhel
-        guest return code is zero,for windows guest,return code is not zero.
+        Start qemu-guest-agent in guest.
         :param session: use for sending cmd
-        :param env: Dictionary with test environment.
+        :param vm: Virtual machine object.
         :param args: Start cmd.
         """
-
         if args and isinstance(args, tuple):
             gagent_start_cmd = args[0]
         else:
-            raise error.TestError("Missing config 'gagent_start_cmd'")
+            self.test.error("Missing config 'gagent_start_cmd'")
 
         if not gagent_start_cmd:
-            return
+            self.test.error("Gagent_start_cmd's value is empty.")
 
-        error_context.context("Try to start 'qemu-guest-agent'.", logging.info)
+        error_context.context("Try to start qemu-ga service.", logging.info)
         s, o = session.cmd_status_output(gagent_start_cmd)
+        # if start a running service, for rhel guest return code is zero,
+        # for windows guest,return code is not zero
         if s and "already been started" not in o:
-            raise error.TestFail("Could not start qemu-guest-agent in VM"
-                                 " '%s', detail: '%s'" % (vm.name, o))
+            self.test.fail("Could not start qemu-ga service in VM '%s',"
+                           "detail: '%s'" % (vm.name, o))
+
+    @error_context.context_aware
+    def gagent_stop(self, session, vm, *args):
+        """
+        Stop qemu-guest-agent in guest.
+        :param session: use for sending cmd
+        :param vm: Virtual machine object.
+        :param args: Stop cmd.
+        """
+        if args and isinstance(args, tuple):
```

Comment:
Just a advice, can "gagent_stop" and "gagent_start" merge to one function, and use gagent_cmd to control it .Since it seems the codes almost the same.

Reply:
Two reasons.
1)separate them to make them clear
2)if condition is not the same, better to separate